components/base-war/src/main/webapp/resources/uicomponents/widgets/userpicker/userPicker.js

```diff
@@ -128,7 +130,7 @@ widgets.UserPicker = Class.create(widgets.Suggest, {
       avatarWrapper.insert(new Element('span', {alt: data.info, 'class': data.icon}));
     }
     container.insert(avatarWrapper);
-    var userName = source.highlight ? this.emphasizeMatches(this.sInput, data.info) : data.info;
+    var userName = source.highlight ? this.emphasizeMatches(this.sInput.escapeHTML(), data.info) : data.info;
```

Comment:
Are you sure the input needs to be escaped, and not the data?

Reply:
Ah, the data is already escaped, right at the top where the big comment is. I guess this is the ALAP escaping point.

Reply:
The data gets escaped in the change above.
Without escaping `sInput`, inputting something like `<img ` does not highlight properly (if I remember correctly).